pilot/pkg/proxy/envoy/v2/config.go

```diff
@@ -744,12 +722,9 @@ func buildInboundListeners(mesh *meshconfig.MeshConfig, node model.Proxy,
 								route.OpaqueConfig = v1.BuildMixerOpaqueConfig(!mesh.DisablePolicyChecks, false,
 									instance.Service.Hostname)
 							}
-						}
-
-						host.Routes = append(host.Routes, routes...)
-					default:
-						panic("unsupported rule")
 					}
+
+					host.Routes = append(host.Routes, routes...)
```

Comment:
I would really like a comment above explaining why we need all the route rules when building listeners. And TODO: to only do the required route processing when building LDS, unless we decide to generate static routes and drop RDS...

Reply:
These are only routes for the proxy.. See a few lines above. config.RouteRulesByDestination(proxy...,) )(I will rename that function)..

Reply:
Not sure what you mean - yes, they are for the sidecar, and we need to get it to work. 

I think we should have another discussion on how 0.8 will be adopted - will we only enable LDSv2 for
users who have no alpha1 ? Can we keep this (with big 'deprecated, will be removed soon' comment) until we have more clarity ? 

Martin and me can work on keeping alpha1 working - we can put all the related code in a 'deprecated.go' file and I know it is throw away, but I would rather waste some time on it and have LDSv2 switched by default sooner and for all users. Once we have confirmation that automatic migration of alpha1 to alpha3 works - we can delete.